### PR TITLE
UL2018 scale and smearing added

### PR DESCRIPTION
**Pull request by jainshilpi**
This is just adding the UL2018 scale and smearing corrections to calibratedEgammas. 
Link to the presentations on this in e/gamma:
https://indico.cern.ch/event/879921/contributions/3825881/attachments/2022015/3381352/ul2018_validationSuite_v7.pdf
https://indico.cern.ch/event/890328/contributions/3801123/attachments/2013185/3364541/ul2018_validationSuite_VF.pdf

The final corrections put in are very close to as shown in the above presentations. The final validation plots using the corrections which are put in can be found here:
https://shilpi-afs.web.cern.ch/shilpi-afs/13TeV/2020/SSvalidation/UL2018/19Oct/

### CODE REVIEW

**Comment by mariadalfonso** [comment]
please test

**Comment by mrodozov** [comment]
abort

**Comment by mrodozov** [comment]
please test with https://github.com/cms-data/EgammaAnalysis-ElectronTools/pull/9

File: RecoEgamma/EgammaTools/python/calibratedEgammas_cff.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 _correctionFile2016Legacy = "EgammaAnalysis/ElectronTools/data/ScalesSmearings/Legacy2016_07Aug2017_FineEtaR9_v3_ele_unc"
 _correctionFile2017Nov17 = "EgammaAnalysis/ElectronTools/data/ScalesSmearings/Run2017_17Nov2017_v1_ele_unc"
 _correctionFile2017UL    = "EgammaAnalysis/ElectronTools/data/ScalesSmearings/Run2017_24Feb2020_runEtaR9Gain_v2"
+_correctionFile2018UL    = "EgammaAnalysis/ElectronTools/data/ScalesSmearings/Run2018_29Sep2020_RunFineEtaR9Gain"
```
Comment (mrodozov) [inline RecoEgamma/EgammaTools/python/calibratedEgammas_cff.py:6]: this data PR seems to be working, I'm merging it

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/31936	71e0a3b 
- code changes are in line with the PR description
- jenkins tests pass and comparisons with the baseline show somewhat minor differences in distributions of slimmed photons and electrons userFloats for the UL 2018 workflow 136.88811, as expected (the right workflow has changes, and offhand I expect mostly smallish changes between 2017 and 2018 for this kind of number of events tested)

@jainshilpi 
please edit the PR description to add a link to some slides or related documentation of the 2018 scale/smearing to improve self-documentation

**Comment by silviodonato** [comment]
+1

> @jainshilpi
please edit the PR description to add a link to some slides or related documentation of the 2018 scale/smearing to improve self-documentation

you should also in the description and title quote that you are changing also 2017

```
from Configuration.Eras.Modifier_run2_egamma_2017_cff import run2_egamma_2017
run2_egamma_2017.toModify(calibratedEgammaSettings,correctionFile = _correctionFile2017UL)
```

**Comment by slava77** [comment]
> you should also in the description and title quote that you are changing also 2017
> 
> ```
> from Configuration.Eras.Modifier_run2_egamma_2017_cff import run2_egamma_2017
> run2_egamma_2017.toModify(calibratedEgammaSettings,correctionFile = _correctionFile2017UL)
> ```

this is more of a no-op: the value without this modifier is the same

**Comment by jainshilpi** [comment]
> +1
> 
> for #31936	[71e0a3b](https://github.com/cms-sw/cmssw/commit/71e0a3b04f32ca25f2224eff6d0821c1f2e5c379)
> 
>     * code changes are in line with the PR description
> 
>     * jenkins tests pass and comparisons with the baseline show somewhat minor differences in distributions of slimmed photons and electrons userFloats for the UL 2018 workflow 136.88811, as expected (the right workflow has changes, and offhand I expect mostly smallish changes between 2017 and 2018 for this kind of number of events tested)
> 
> 
> @jainshilpi
> please edit the PR description to add a link to some slides or related documentation of the 2018 scale/smearing to improve self-documentation

@slava77 Updated the PR description. I will now do a backport of this in 10_6_X .

**Pull request merged**